modules/ocf_backups/files/prune-old-backups

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import json
+import shlex
+import subprocess
+import sys
+import textwrap
+
+RETENTION_TIME = datetime.timedelta(days=180) # 6 months
+
+BOX_FTP = 'ftps://ftp.box.com:990'
+BACKUP_PREFIX = 'ocf-backup-'
+
+LFTP_INIT = '''
+    set ftps:initial-prot ""
+    set ftp:ssl-force true
+    set ftp:ssl-protect-data true
+    open {box_ftp}
+    user {email} {passwd}
+'''
+
+def parse_date(dtstr):
+    """Converts a 'YYYY-MM-DD' formatted string to a datetime.date object."""
+    if len(dtstr) != 10:
+        raise ValueError
+
+    year = int(dtstr[0:4])
+
+    if dtstr[4] != '-':
+        raise ValueError
+
+    month = int(dtstr[5:7])
+
+    if dtstr[7] != '-':
+        raise ValueError
+
+    day = int(dtstr[8:10])
+
+    return datetime.date(year, month, day)
+
+def list_items(creds):
+    """List all items at the root of the Box.com folder."""
+    lftp_cmd = (LFTP_INIT + 'renlist').format(
+        box_ftp=BOX_FTP,
```

Comment:
I would abstract these out into another function like `lftp_cmd(command, output=subprocess.PIPE)`

Reply:
There are enough differences between the two command calls that I think it more sense to keep them separate. An abstraction here would just make the code more confusing.

Reply:
it looks like the major difference is `renlist` vs `rm -r {items}.format(items=' '.join(shlex.quote...)`? unless I'm missing something?

would 

```python
def lftp_cmd(command, out=subprocess.PIPE):
  cmd = LFTP_INIT.format(...)
  cmd += command
  p = subprocess.run(('lftp', '--norc'), input=cmd, output=out)
  p.check_returncode()
```

work?

Reply:
The format args dict has to get passed in both times, but that breaks the abstraction barrier because the caller has to know which template fields to fill in for `LFTP_INIT`.

Reply:
I mean both are filling in `box_ftp`, `email`, and `passwd`, and the command format field can be rendered by the caller, right? e.g.

`lftp_cmd('renlist')` and `lftp_cmd('rm -r ' + ' '.join(shlex.quote(item) for item in items))`

Reply:
Ok on second thought you're right, since this makes the `list_items` and `delete_items` functions clearer to understand. New commit incoming